### ISSUE TITLE
WebGPU support detection

### ISSUE THREAD

**Issue by ekharkunov** [defold/defold#9594]
**Task (REQUIRED):**
Current detection of what graphics backend should use is not working correctly.
First problem - wrong class compared to when check WebGPU support on JS side https://github.com/defold/defold/blob/a272ebeb68af9b73095ecb413a3248a5dc389ef4/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js#L809. Should be https://developer.mozilla.org/en-US/docs/Web/API/GPUCanvasContext

Second problem - WebGPU selected as initial backend without check whether supported or not. Possible fix and full steps to reproduce https://github.com/defold/defold/pull/9585#discussion_r1794965277

**Expected outcome (REQUIRED):**
WebGPU detects and works correctly without any errors in the console. And WebGL use as fallback backend if WebGPU is not supported.

**Comment by AGulev** [comment]
is it something you fixed a while back?

**Comment by ekharkunov** [comment]
> is it something you have fixed?

No. Fixed in https://github.com/defold/defold/pull/9947 and https://github.com/defold/defold/pull/10381
